app/models/submission_similarity.rb

```diff
@@ -55,4 +55,8 @@ def other_student(student)
   def other_submission(student)
     (student == submission1.student) ? submission2 : submission1
   end
+
+  def similarity_mappings
+    SubmissionSimilarityMapping.where("submission_similarity_id = ? AND statement_count > 0", self.id)
```

Comment:
Probably no problem, but could any of these statements be a problem if the course and mapping files are large?

Reply:
It just retrieved line mappings of a single comparison between 2 submissions, not all pair-wise comparison. So I think it would not cause problems.